tests/utils/test_env.py

```diff
@@ -0,0 +1,25 @@
+from poetry.utils.env import Env
+from poetry.utils._compat import Path
+import sys
+
+
+def test_execute_windows_calls_exe(mocker):
+    sys.platform = "win32"
```

Comment:
I don't think changing `sys.platform` is a good idea since it will stick for the reset of the tests. It's better in this case to mock `Env._is_windows`.